larpandoracontent/LArTrackShowerId/SvmPfoCharacterisationAlgorithm.cc

```diff
@@ -100,14 +100,26 @@ bool SvmPfoCharacterisationAlgorithm::IsClearTrack(const pandora::ParticleFlowOb
         return isTrueTrack;
     }// training mode
 
-    if (!m_enableProbability)
-    {
-        return LArSvmHelper::Classify((wClusterList.empty() ? m_supportVectorMachineNoChargeInfo : m_supportVectorMachine), featureVector);
-    }
-    else
-    {
-        return (m_minProbabilityCut <= LArSvmHelper::CalculateProbability((wClusterList.empty() ? m_supportVectorMachineNoChargeInfo : m_supportVectorMachine), featureVector));
-    }
+	//check for failures in the calculation of features, which would give -1 in any of the features, if that happens, use current id
+	bool useSVM(true);
+	for (int i=0; i<featureVector.size(); i++)
+		if (std::fabs(featureVector[i] + 1.f) < std::numeric_limits<float>::epsilon())
+				useSVM = false;
```

Comment:
Maybe better just removing the useSVM variable and just setting the value and returning immediately if this check succeeds and you can't proceed with the SVM as normal.

Also, need some parentheses here, and the indentation looks non-standard (at least in browser on GitHub, it does).

Finally, prefer e.g. `for (const double featureValue : featureVector)`, rather than use of operator[]